### PR TITLE
Test case that fails due to NullPointerException

### PR DESCRIPTION
**Pull request by suztomo**
Investigating the reason for https://github.com/apache/beam/pull/14527#discussion_r613980011

### CODE REVIEW

**Review by elharo — COMMENTED** [review]
The mocking makes it hard to figure out whether there's a bug in the model code or just a bad mock setup. See https://github.com/googleapis/google-http-java-client/pull/1345 for a test that uses a real HTTP server instead of a mock.

**Review by elharo — COMMENTED** [review]
Possibly BatchUnparsedResponse needs to handle empty input better

**Comment by suztomo** [comment]
I agree that this test case is not clear which one is the problem. I'll try to simplify this further.

File: google-http-client-jackson2/src/test/java/com/google/api/client/json/jackson2/BatchTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+    batch.execute();
+    assertNotNull(capturedGoogleJsonError[0]);
+
+    // From {"error":{"code":404}}
+    assertEquals(404, capturedGoogleJsonError[0].getCode());
+  }
+
+  @Test
+  public void testErrorContentRead_NoRetry() throws IOException {
```
Comment (suztomo) [inline google-http-client-jackson2/src/test/java/com/google/api/client/json/jackson2/BatchTest.java:148]: This succeeds. Maybe problem in retry?

**Review by elharo — COMMENTED** [review]
I don't think this test demonstrates a bug or an issue with the http client. The NullPointerException here occurs in google-api-java-client. The code there looks very suspicious to me.

If there is an issue here it should be able to be demonstrated with classes already in this repository.

File: google-http-client-jackson2/src/test/java/com/google/api/client/json/jackson2/BatchTestWithFakeServer.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+              byte[] response = content.getBytes(StandardCharsets.UTF_8);
+              responseHeaders.set("Content-Type", "multipart/mixed; boundary=" + contentBoundary);
+              httpExchange.sendResponseHeaders(200, response.length);
+              try (OutputStream out = httpExchange.getResponseBody()) {
+                out.write(response);
+              }
+            }
+          }
+        };
+    try (FakeServer server = new FakeServer(handler)) {
```
Comment (suztomo) [inline google-http-client-jackson2/src/test/java/com/google/api/client/json/jackson2/BatchTestWithFakeServer.java:105]: With the fake server, this test succeeds. This implies that the mock setup in Beam is not enough (at least for this test case).

**Comment by suztomo** [comment]
The FakeServer in #1345 worked. Thank you. Now I'm feeling something is lacking in the mock setup.

> If there is an issue here it should be able to be demonstrated with classes already in this repository

Right. I haven't been able to isolate the problem in a class yet.

**Comment by suztomo** [comment]
Next step: investigate why mock fails and fake server succeeds.

**Pull request closed without merging**